save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/TestSuiteSourceCreationComponent.kt

```diff
@@ -0,0 +1,218 @@
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.testsuite.TestSuitesSourceDto
+import com.saveourtool.save.v1
+
+import csstype.ClassName
+import react.ChildrenBuilder
+import react.FC
+import react.Props
+import react.dom.aria.ariaLabel
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.h5
+import react.useState
+
+import kotlin.js.json
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+val testSuiteSourceCreationComponent = testSuiteSourceCreationComponent()
+
+@Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
+private val selectFormRequired = selectFormRequired<GitDto>()
+
+/**
+ * [Props] for [testSuiteSourceCreationComponent]
+ */
+external interface TestSuiteSourceCreationProps : Props {
+    /**
+     * Name of a current organization
+     */
+    var organizationName: String
+
+    /**
+     * Callback invoked on successful save
+     */
+    var onSuccess: () -> Unit
+}
+
+/**
+ * @param isOpen
+ * @param organizationName
+ * @param onSuccess
+ * @param onClose
+ */
+@Suppress("TOO_LONG_FUNCTION")
+fun ChildrenBuilder.showTestSuiteSourceCreationModal(
+    isOpen: Boolean,
+    organizationName: String,
+    onSuccess: () -> Unit,
+    onClose: () -> Unit,
+) {
+    modal { props ->
+        props.isOpen = isOpen
+        props.style = Styles(
+            content = json(
+                "top" to "10%",
+                "left" to "30%",
+                "right" to "30%",
+                "bottom" to "auto",
+                "position" to "absolute",
+                "overflow" to "hide",
+            ).unsafeCast<CssProperties>()
+        )
+        div {
+            className = ClassName("d-flex justify-content-between")
+            h5 {
+                className = ClassName("modal-title mb-3")
+                +"Create test suite source"
+            }
+            button {
+                type = ButtonType.button
+                className = ClassName("close")
+                asDynamic()["data-dismiss"] = "modal"
+                ariaLabel = "Close"
+                fontAwesomeIcon(icon = faTimesCircle)
+                onClick = {
+                    onClose()
+                }
+            }
+        }
+        div {
+            testSuiteSourceCreationComponent {
+                this.organizationName = organizationName
+                this.onSuccess = onSuccess
+            }
+        }
+    }
+}
+
+@Suppress("TOO_LONG_FUNCTION", "LongMethod")
+private fun testSuiteSourceCreationComponent() = FC<TestSuiteSourceCreationProps> { props ->
+    val (testSuiteSource, setTestSuiteSource) = useState(TestSuitesSourceDto.empty.copy(organizationName = props.organizationName))
+    val (saveStatus, setSaveStatus) = useState<SourceSaveStatus?>(null)
+    val onSubmitButtonPressed = useRequest(dependencies = arrayOf(testSuiteSource)) {
```

Comment:
Does deferred request need dependencies?

Reply:
Oh, from the first glance it seemed to work as planned - sending request only if dependencies are changed, but not I am not so sure. Seems there is no need in it.

Reply:
Now that I look at `useRequest` I don't understand why we need dependencies here:
https://github.com/saveourtool/save-cloud/blob/b665d1d8e54b3ec2ed5607d2ee38f8241db61f89/save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt#L333

It makes sense to only use `isSending` when request is deferred and needs to only be activated by a trigger, e.g. a button click. And to use `dependencies` only in the effect, which is returned if `isDeferred  == false`:
https://github.com/saveourtool/save-cloud/blob/b665d1d8e54b3ec2ed5607d2ee38f8241db61f89/save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/RequestUtils.kt#L354-L358

Seems like now change of one of dependencies causes many invocations of the first effect, which are effectively negated by `isSending` being false

Reply:
It also seems a bit weird that there actually are lots of `useEffect` calls (on line 333 and so on). Is there any other way to escape this cause it seems it can cause bit problems with efficiency.

Reply:
`useEffect` is used for callable request too, because it allows us to control the CoroutineScope and cancel it when component unmounts. But indeed it is not necessary to wrap one effect into another when a request is not deferred.

Reply:
The fun thing is that if there is only `props.organizationName` in dependencies and I invoke `fetchTestSuitesSources()`, table in not updated with a new created source. If I add a trigger-flag into dependencies of useRequest, then I do not even need to invoke `fetchTestSuitesSources()`, which is fun. Will remove invocation and just leave flag

Reply:
There is no need in dependencies if `isDeferred` is true, because `useRequest` contains a trigger flag inside

Reply:
Fun thing is that nothing happens if I call `fetchTestSuitesSources()` after source is created